README.md

```diff
@@ -151,6 +145,25 @@ in the build directory.
 
 To build the Python API see [Python.md](Python.md).
 
+Compiling in Docker
+-------------------
+
+Instead of setting up a development environment explicitly you can build Katana in docker.
+
+```Shell
+scripts/build_in_container.py -B $BUILD_DIR --type conda
+```
+where `$BUILD_DIR` is a path at which to place the resulting build directory.
+Build types other than `conda` may be supported in the future.
+You can also pass build targets to the command.
+
+For example,
+```Shell
+scripts/build_in_container.py -B ~/katana-build --type conda docs
```

Comment:
Are you purposefully obfuscating the `docker run` command? I personally think, even with a python/make command, we should still provide an example docker run command that someone can use. Not a hard requirement.

Reply:
Because this is designed to run from the source tree, I chose to have a script that *builds* the docker image on the users machine and then runs it. The main advantage of this is that the docker image cannot be out of date with respect to the build dependencies specified in the repo and no one needs to upload new image versions. The user needs to have the source tree to do the build anyway, so I don't see much of a downside in this approach.

I can provide a docker command to rerun the container that is built by this script if that's useful. But it's an irritatingly complex command because there are 2 mounts and a couple of environment variables. Unless you have a specific request on this, I'll make the script print the command the user can use when it completes. Then the user can manually run things if they like after the script builds the image.

Reply:
Okay, no problem. My only concern is on documentation edit or source code edit, whether the build time significant enough to be an issue.

Reply:
Docker build caches intermediates so it should be fast after the first run. Also there is a `--reuse` option on the command that causes it to reuse the image without rebuilding it at all. But I didn't advertise it because it *forces* reuse even if the image is out of date and there is no easy way to avoid that.

Reply:
I improved the output a bit and added some options to hopefully make this clearer for anyone using the script.